src/wsrep_provider_v26.cpp

```diff
@@ -690,6 +690,29 @@ wsrep::wsrep_provider_v26::run_applier(
     return map_return_value(wsrep_->recv(wsrep_, applier_ctx));
 }
 
+enum wsrep::provider::status
+wsrep::wsrep_provider_v26::assign_read_view(wsrep::ws_handle& ws_handle,
+                                            const wsrep::gtid* gtid)
+{
+    const wsrep_gtid_t* gtid_ptr(NULL);
+    wsrep_gtid_t tmp;
+    if (sizeof(wsrep::gtid) == sizeof(wsrep_gtid_t))
```

Comment:
How do you know that the internal layout of the wsrep::gtid matches to wsrep_gtid_t? I would get rid of this and always copy data to tmp wsrep_gtid_t.